staging/src/k8s.io/apiserver/pkg/util/shufflesharding/shufflesharding.go

```diff
@@ -87,11 +101,33 @@ func ShuffleAndDealToSlice(hashValue uint64, deckSize, handSize int32) []int32 {
 	return candidates
 }
 
+// ShuffleAndDealIntoHand shuffles a deck of the given size by the
+// given hash value and deals cards into the given slice.  The virtue
+// of this function compared to ShuffleAndDealToSlice is that the
+// caller provides the storage for the hand.
+func ShuffleAndDealIntoHand(hashValue uint64, deckSize int32, hand []int) {
+	handSize := int32(len(hand))
+	var idx int
+	ShuffleAndDeal(hashValue, deckSize, handSize, func(card int32) {
+		hand[idx] = int(card)
```

Comment:
isn't idx always 0, since ShuffleAndDeal is called only once inside ShuffleAndDealIntoHand?

Reply:
`ShuffleAndDeal` calls its `pick` parameter (which is the function here) `handSize` times, and each increments `idx`

Reply:
I think Mike is right.